app/parse.py

```diff
@@ -1,16 +1,124 @@
+import csv
 from dataclasses import dataclass
 
+import requests
+from bs4 import BeautifulSoup
+
+
+BASE_URL = "https://quotes.toscrape.com/"
+
+
 
 @dataclass
 class Quote:
     text: str
     author: str
     tags: list[str]
+    author_url: str
+
+
+@dataclass
+class Author:
+    name: str
+    biography: str
+    birth_date: str
+    birth_place: str
+
+
+def parse_single_page(page_content: str) -> list[Quote]:
+    soup = BeautifulSoup(page_content, "html.parser")
+    quotes = []
+    for quote in soup.select(".quote"):
+        author_url = (BASE_URL +
+                      quote.select_one("a[href^='/author/']").get("href")
+                      )
+        quotes.append(Quote(
+            text=quote.select_one(".text").text,
+            author=quote.select_one(".author").text,
+            tags=[tag.text for tag in quote.select(".tag")],
+            author_url=author_url,
+        ))
+    return quotes
+
+
+def get_num_pages() -> int:
+    page_number = 1
+
+    while True:
+        page = requests.get(f"{BASE_URL}page/{page_number}")
+        soup = BeautifulSoup(page.content, "html.parser")
+        if not soup.select_one(".next"):
+            break
+        page_number += 1
+    return page_number
+
+def get_author_biography(author_url: str) -> Author:
+    page = requests.get(author_url)
```

Comment:
Consider adding error handling for the HTTP request to manage potential network issues or invalid responses.